src/documents.rs

```diff
@@ -1,6 +1,16 @@
-use crate::{errors::Error, indexes::Index};
+use async_trait::async_trait;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 
+use crate::settings::Settings;
+use crate::tasks::Task;
+use crate::{errors::Error, indexes::Index};
+
+#[async_trait]
+pub trait Document {
+    fn generate_settings() -> Settings;
+    async fn generate_index(client: &crate::client::Client) -> Result<Index, Task>;
+}
```

Comment:
Since this trait can be derived using a derive proc macro, we should also re-export and document the macro here. You can add the following snippet above the trait declaration, and maybe improve the documentation I came up with if you want :)
```rust

/// Derive the [`Document`](crate::documents::Document) trait.
///
/// ## Field attribute
/// Use the `#[document(..)]` field attribute to generate the correct settings
/// for each field. The available parameters are:
/// - `primary_key` (can only be used once)
/// - `distinct` (can only be used once)
/// - `searchable`
/// - `displayed`
/// - `filterable`
/// - `sortable`
///
/// ## Index name
/// The name of the index will be the name of the struct converted to snake case.
///
/// ## Sample usage:
/// ```
/// use serde::{Serialize, Deserialize};
/// use meilisearch_sdk::documents::Document;
/// use meilisearch_sdk::settings::Settings;
/// use meilisearch_sdk::indexes::Index;
/// use meilisearch_sdk::client::Client;
///
/// #[derive(Serialize, Deserialize, Document)]
/// struct Movie {
///     #[document(primary_key)]
///     movie_id: u64,
///     #[document(displayed, searchable)]
///     title: String,
///     #[document(displayed)]
///     description: String,
///     #[document(filterable, sortable, displayed)]
///     release_date: String,
///     #[document(filterable, displayed)]
///     genres: Vec<String>,
/// }
///
/// async fn usage(client: Client) {
///     // Default settings with the distinct, searchable, displayed, filterable, and sortable fields set correctly.
///     let settings: Settings = Movie::generate_settings();
///     // Index created with the name `movie` and the primary key set to `movie_id`
///     let index: Index = Movie::generate_index(&client).await.unwrap();
/// }
/// ```
pub use meilisearch_index_setting_macro::Document;
```

Reply:
This is throwing an error:

```
error[E0433]: failed to resolve: unresolved import
  --> src/documents.rs:27:34
   |
10 | #[derive(Serialize, Deserialize, Document)]
   |                                  ^^^^^^^^
   |                                  |
   |                                  unresolved import
   |                                  help: a similar path exists: `meilisearch_sdk::client`
   |
   = note: this error originates in the derive macro `Document` (in Nightly builds, run with -Z macro-backtrace for more info)
```